lib/ansible/modules/cloud/ovirt/ovirt_storage_domains.py

```diff
@@ -501,10 +518,19 @@ def main():
         state = module.params['state']
         control_state(storage_domains_module)
         if state == 'absent':
+            # Pick random available host when host parameter is missing
+            host_param = module.params['host']
+            if not host_param:
+                hosts_service = connection.system_service().hosts_service()
```

Comment:
We need to make sure that the host is in the same data center as the storage.

Reply:
Thanks, I 'll repost

Reply:
I think you can use:

```python
hosts_service = connection.system_service().hosts_service(search="storage=module.param['storage']")
```

Reply:
Here is some more info:

https://www.ovirt.org/documentation/admin-guide/appe-Using_Search_Bookmarks_and_Tags/

Reply:
BTW I think we need to make sure here that the host will be in the same cluster as the SD, not Datacenter.. In this case, we can also choose correct host, even in the case that SD is detached from the Datacenter.

Reply:
SD is per DC, so cluster will work as well, but still you don't have `cluster` param specified by user, only SD.

Reply:
Although this command looked promising I think it works only for attached storages.

What I mean, is that if I call the module with state == absent  and no host parameter set, with some storage domain name which is already unattached (thus I want removal step done), this command doesn't return any hosts for me. :/
Also I read the documentation for Searching for hosts, and `storage search` doesn't look to be supported.

I 'll find some other WA for this.